<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="assy_part_07">
  <xacro:macro name="assy_part_07" params="prefix parent *origin">
    <link name="${prefix}assy_part_07">
      <visual>
        <origin rpy="1.5707 0 0" xyz="0 0 0"/>
        <geometry>
          <mesh filename="package://o2as_parts_description/meshes/07_SBARB6200ZZ_30.dae" scale="0.001 0.001 0.001"/>
        </geometry>
        <material name="white">
          <color rgba="1 1 1 1"/>
        </material>
      </visual>
      <collision>
        <origin rpy="1.5707 0 0" xyz="0 0 0"/>
        <geometry>
          <mesh filename="package://o2as_parts_description/meshes/07_SBARB6200ZZ_30.vhacd.dae" scale="0.001 0.001 0.001"/>
        </geometry>
      </collision>
    </link>

    <joint name="${prefix}assy_part_07_joint" type="fixed">
      <parent link="${parent}"/>
      <child link="${prefix}assy_part_07"/>
      <xacro:insert_block name="origin"/>
    </joint>
  
  <!-- Extra frames for assembly and motion planning, if any -->
    <joint name="${prefix}assy_part_07_front_hole_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_front_hole"/> 
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_front_hole"/> 
    
    <joint name="${prefix}assy_part_07_back_hole_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_back_hole"/> 
      <origin rpy="0.0 0.0 ${pi}" xyz="-0.03 0.0 0.0 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_back_hole"/> 
    
    <joint name="${prefix}assy_part_07_back_tip_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_back_tip"/> 
      <origin rpy="0.0 0.0 0.0" xyz="-0.03 0.0 0.0 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_back_tip"/> 
    
    <joint name="${prefix}assy_part_07_inserted_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_inserted"/> 
      <origin rpy="0.0 0.0 0" xyz="-0.008 0.0 0.0 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_inserted"/> 
    
    <joint name="${prefix}assy_part_07_screw_hole_1_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_screw_hole_1"/> 
      <origin rpy="0.0 0.0 ${pi}" xyz="0.0 0.022 0.0 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_screw_hole_1"/> 
    
    <joint name="${prefix}assy_part_07_screw_hole_2_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_screw_hole_2"/> 
      <origin rpy="0.0 0.0 ${pi}" xyz="0.0 -0.022 0.0 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_screw_hole_2"/> 
    
    <joint name="${prefix}assy_part_07_screw_hole_3_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_screw_hole_3"/> 
      <origin rpy="0.0 0.0 ${pi}" xyz="0.0 0.0 0.022 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_screw_hole_3"/> 
    
    <joint name="${prefix}assy_part_07_screw_hole_4_joint" type="fixed"> 
      <parent link="${prefix}assy_part_07"/> 
      <child link="${prefix}assy_part_07_screw_hole_4"/> 
      <origin rpy="0.0 0.0 ${pi}" xyz="0.0 0.0 -0.022 "/> 
    </joint> 
    <link name="${prefix}assy_part_07_screw_hole_4"/> 
    

  </xacro:macro>
</robot>
